### ISSUE TITLE
dd.Series.isin() broken for dict views in distributed

### ISSUE THREAD

**Issue by gsakkis** [dask/dask#4974]
#4727 inadvertently broke `isin` for dict views (`.keys()`, `.values()`, `.items()`) and potentially other iterables when running with the distributed scheduler; more details [here](https://github.com/dask/dask/pull/4727#issuecomment-497635280). cc @jcrist

**Comment by TomAugspurger** [comment]
And quoting a few more things.

> does it make since to include iterators like dict_keys and dict_values in dask.delayed's `unpack_collections`?

Reproducer

```diff
diff --git a/dask/dataframe/tests/test_dataframe.py b/dask/dataframe/tests/test_dataframe.py
index 0e18f510..d82701ef 100644
--- a/dask/dataframe/tests/test_dataframe.py
+++ b/dask/dataframe/tests/test_dataframe.py
@@ -932,6 +932,21 @@ def test_isin():
             d.isin(obj)
 
 
+def test_isin_distributed():
+    import distributed
+    client = distributed.Client()
+
+    b_dict = {1: 'foo', 2: 'bar'}
+
+    # these work
+    d1 = d.isin(set(b_dict)).compute()
+    d2 = d[d['b'].isin(set(b_dict))].compute()
+
+    # these hang on dask>=1.2.1
+    d3 = d.isin(b_dict.keys()).compute()
+    d4 = d[d['b'].isin(b_dict.keys())].compute()
+
+
 def test_len():
     assert len(d) == len(full)
     assert len(d.a) == len(full.a)

```

**Comment by martindurant** [comment]
Since there is no activity here, I suggest putting the above failing test into a PR and discussing there how it should be fixed. @gsakkis , would you like to do that?

**Comment by gsakkis** [comment]
You mean a PR with just the (failing) test initially? Yeah I could do that. Though since it's hanging instead of   raising an error we'd need a way to timebox it, is there such a utility in the tests?

**Comment by martindurant** [comment]
There certainly are over in distributed (where you can expect async operations), not sure in Dask. It's OK if it hangs, though, or you could even add [skip ci] for no testing of the initial commit; however, it may be that this actually belongs in distributed rather than here.

**Comment by gsakkis** [comment]
Indeed it's not clear where it belongs; on the one hand it manifests with distributed only, on the other hand it was introduced with a change in dask. For now I opened #5009 here.

While looking for how to timebox it, I modified the original test to call [client.compute](https://distributed.readthedocs.io/en/latest/api.html#distributed.Client.compute) instead of [DataFrame.compute](https://docs.dask.org/en/latest/dataframe-api.html#dask.dataframe.compute); the former returns Futures which can then be [waited](https://distributed.readthedocs.io/en/latest/api.html#distributed.wait) with a timeout. What's interesting is that with this change the test fails with a `TypeError` ('Could not serialize object of type dict_keys') instead of timeout error. Probably useful to know for debugging it.
